brave-instrumentation/aws-java-sdk-v2-core/src/test/java/brave/instrumentation/awsv2/ITTracingExecutionInterceptor.java

```diff
@@ -0,0 +1,314 @@
+    get(client, uri);
+
+    List<Span> spans = asList(takeSpan(), takeSpan());
+    spans.stream().filter(s -> s.parentId() != null).forEach(s -> {
+      assertThat(s.remoteServiceName())
+          .isEqualTo("dynamodb");
+      assertThat(s.name()).isEqualTo("getitem");
+
+      assertThat(s.tags())
+          .containsEntry("http.url", url(uri))
+          .containsEntry("request_customizer.is_span", "false")
+          .containsEntry("response_customizer.is_span", "false");
+    });
+  }
+
+  /*
+   * Tests overriden due to the multi-span implementation of the client, application and client
+   */
+
+  @Override public void usesParentFromInvocationTime() throws Exception {
+    Tracer tracer = httpTracing.tracing().tracer();
+    server.enqueue(new MockResponse().setBodyDelay(300, TimeUnit.MILLISECONDS));
+    server.enqueue(new MockResponse());
+
+    ScopedSpan parent = tracer.startScopedSpan("test");
+    try {
+      getAsync(client, "/items/1");
+      getAsync(client, "/items/2");
+    } finally {
+      parent.finish();
+    }
+
+    ScopedSpan otherSpan = tracer.startScopedSpan("test2");
+    List<String> requestSpanIds = new ArrayList<>();
+    try {
+      for (int i = 0; i < 2; i++) {
+        RecordedRequest request = server.takeRequest();
+        assertThat(request.getHeader("x-b3-traceId"))
+            .isEqualTo(parent.context().traceIdString());
+        requestSpanIds.add(request.getHeader("x-b3-spanId"));
+      }
+    } finally {
+      otherSpan.finish();
+    }
+
+    // Check we reported 2 in-process spans, 2 sdk spans, and 2 RPC client spans
+    List<Span> spans = asList(takeSpan(), takeSpan(), takeSpan(), takeSpan(), takeSpan(), takeSpan());
+
+    // Get request spans, then get their parents (sdk spans), then make sure the sdk spans have the
+    // correct parent
+    List<String> sdkSpanIds = spans.stream().filter(s -> requestSpanIds.contains(s.id()))
+        .map(Span::parentId).collect(Collectors.toList());
+    spans.stream().filter(s -> sdkSpanIds.contains(s.id())).forEach(s -> {
+      assertThat(s.parentId()).isEqualToIgnoringCase(HexCodec.toLowerHex(parent.context().spanId()));
+    });
+
+
+    assertThat(spans)
+        .extracting(Span::kind)
+        .containsOnly(null, Span.Kind.CLIENT);
+  }
+
+  /** Modified to check hierarchy from parent->application->client */
+  @Override public void makesChildOfCurrentSpan() throws Exception {
+    Tracer tracer = httpTracing.tracing().tracer();
+    server.enqueue(new MockResponse());
+
+    ScopedSpan parent = tracer.startScopedSpan("test");
+    try {
+      get(client, "/foo");
+    } finally {
+      parent.finish();
+    }
+
+    List<Span> spans = asList(takeSpan(), takeSpan(), takeSpan());
+    Span applicationSpan = spans.stream().filter(s -> HexCodec.toLowerHex(parent.context().spanId()).equals(s.parentId())).findFirst().get();
+    Span clientSpan = spans.stream().filter(s -> applicationSpan.id().equals(s.parentId())).findFirst().get();
```

Comment:
The changes to the above 2 lines need to be back ported. I ran into NPEs when running these that I'm surprised I didn't see last time